tests/unit/amo/components/TestAddonMoreInfo.js

```diff
@@ -551,6 +563,232 @@ describe(__filename, () => {
+    const results = [
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Alerts & Update',
+        slug: 'alert-update',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Blogging',
+        slug: 'blogging',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Games',
+        slug: 'Games',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_FIREFOX,
+        name: 'Alerts & Update',
+        slug: 'alert-update',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_FIREFOX,
+        name: 'Security',
+        slug: 'security',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_FIREFOX,
+        name: 'Anime',
+        slug: 'anime',
+        type: ADDON_TYPE_STATIC_THEME,
+      },
+    ];
+
+    const addon = createInternalAddonWithLang({
+      ...fakeAddon,
+      categories: {
+        [CLIENT_APP_FIREFOX]: [],
+      },
+      type: ADDON_TYPE_EXTENSION,
+    });
+
+    dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+    store.dispatch(loadCategories({ results }));
+
+    const root = render({
+      addon,
+      store,
+    });
+
+    expect(
+      root.find('.AddonMoreInfo-related-categories').find(Link),
+    ).toHaveLength(0);
+  });
+
+  it('does not render related categories when categories for add-on do not exist', () => {
+    const results = [
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Alerts & Update',
+        slug: 'alert-update',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Blogging',
+        slug: 'blogging',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Games',
+        slug: 'Games',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_ANDROID,
+        name: 'Security',
+        slug: 'security',
+        type: ADDON_TYPE_EXTENSION,
+      },
+      {
+        ...fakeCategory,
+        application: CLIENT_APP_FIREFOX,
+        name: 'Anime',
+        slug: 'anime',
+        type: ADDON_TYPE_STATIC_THEME,
+      },
+    ];
+
+    const addon = createInternalAddonWithLang({
+      ...fakeAddon,
+      categories: {
+        [CLIENT_APP_FIREFOX]: ['alert-update', 'security'],
+      },
+      type: ADDON_TYPE_EXTENSION,
+    });
+
+    dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+    store.dispatch(loadCategories({ results }));
```

Comment:
I suggest you load an empty list of categories here or a list with only 1 category.

Reply:
In this case, you dispatch `CLIENT_APP_FIREFOX` and the category `alert-update` is for `CLIENT_APP_ANDROID`. This is good and it is covering the case where a category does not exist for `clientApp`.

I suggest we had another test case with a category for `CLIENT_APP_ANDROID` and a slug like `does-not-exist`, and no `setClientApp()` action dispatched (so that the clientApp is android). The add-on should have categories for android but not `does-not-exist`. This will cover the case where we have categories loaded, categories defined on the add-on, the same client app everywhere but still no category to display.

Reply:
I added a test in the case where there are no loaded categories (empty list of categories).

> In this case, you dispatch `CLIENT_APP_FIREFOX` and the category `alert-update` is for `CLIENT_APP_ANDROID`. This is good and it is covering the case where a category does not exist for `clientApp`.

For this case where a category does not exist for `clientApp`, I updated `alert-update` to `blogging, games` so I could still use the already defined `categories`.

> I suggest we had another test case with a category for `CLIENT_APP_ANDROID` and a slug like `does-not-exist`, and no `setClientApp()` action dispatched (so that the clientApp is android). The add-on should have categories for android but not `does-not-exist`. This will cover the case where we have categories loaded, categories defined on the add-on, the same client app everywhere but still no category to display.

I am not sure if I interpreted this test case correctly, please refer to the test case that has been added on line 710. Is it to test that it only renders related categories if the categories exist for `CLIENT_APP_ANDROID`?

Reply:
> I am not sure if I interpreted this test case correctly, please refer to the test case that has been added on line 710. Is it to test that it only renders related categories if the categories exist for `CLIENT_APP_ANDROID`?

Yeah, it looks quite good but why do we have 2 links instead of 0?

Reply:
I fixed the test case and it looks good now, thanks!